### ISSUE TITLE
Is probability calibration needed?

### ISSUE THREAD

**Issue by JohnZhihongWang** [cgnorthcutt/cleanlab#75]
I am trying to use your cleanlab package in my model. I am using XGBoost to predict probabilities for binary classification. Do I need to do probability calibration on the predicted probabilities (e.g. https://scikit-learn.org/stable/modules/calibration.html) before calling get_noise_indices?

**Comment by cgnorthcutt** [comment]
Hi John,

A question for you, what motivates this question? Have you seen other works with noisy labels require this?

For `cleanlab`, calibration beforehand is not required. `cleanlab` builds on confident learning (see paper: https://www.jair.org/index.php/jair/article/view/12125) which guarantees robustness to several forms of biases and error in model outputs (e.g. if the model is overconfident for certain classes). 

In short, you don't need to use calibration. It hasn't been published yet, but it turns out that confident learning (and therefore cleanlab) actually calibrates for you when working with datasets with noisy labels. You can still use your favorite calibration beforehand if you like -- (maybe it improves the results some, maybe not), but I wouldn't expect much difference because `cleanlab` finds errors based on per-class-thresholds which takes into account whether the model  is over-confident or under-confident for each class.

**If you (or anyone else reading this) finds that calibration improves cleanlab, please post the benchmarks here!**

**Comment by JohnZhihongWang** [comment]
Hi Curtis,

Thanks for the detailed explanation. I did not see other works with noisy labels requiring calibration. In general, calibration may make the probabilities outcome of classification algorithms align better with actual probabilities. That's the reason I am wondering whether un-calibrated or calibrated probabilities work better with cleanlab. It's good to know that an extra calibration step is not needed. Saves a lot of computation.

**Comment by cgnorthcutt** [comment]
Yup you should be good. To further clarify.. cleanlab doesn't actually require probabilities at all (it also works with the raw logits).

Although it's not hard to apply softmax to your logits beforehand, it's not necessary for confident learning to work. However, most people use probabilities (soft max applied to logits) when using cleanlab because it makes the code and data more interpretable.

**Comment by jwmueller** [comment]
For folks ending up on this page recently, this message was referring to old version of the package, which just did one thing at the time. Since then tons of new functionality has been added to the package, which assumes what you are inputting is probabilities not something else like logits
